backend/fms_core/template_importer/importers/sample_add_metadata.py

```diff
@@ -0,0 +1,55 @@
+from ._generic import GenericImporter
+from fms_core.template_importer.row_handlers.sample_add_metadata import SampleAddMetadataHandler
+from fms_core.templates import SAMPLE_ADD_METADATA_TEMPLATE
+
+METADATA_STARTING_INDEX = 4
+
+class SampleAddMetadataImporter(GenericImporter):
+    SHEETS_INFO = SAMPLE_ADD_METADATA_TEMPLATE['sheets info']
+
+    def __init__(self):
+        super().__init__()
+        self.metadata_starting_index = METADATA_STARTING_INDEX
+
+    def import_template_inner(self):
+        """
+            METADATA SHEET
+        """
+        metadata_sheet = self.sheets['Metadata']
+
+        # Add the extra headers to the template sheet according to the metadata
+        metadata_headers = [key for i, (key, val) in enumerate(metadata_sheet.rows[0].items()) if i >= self.metadata_starting_index]
+        extra_headers = {
+            # sheet_number : extra metadata headers
+            0: metadata_headers
+        }
+        self.add_dynamic_headers(extra_headers)
```

Comment:
This appears to be redundant ... I think it may very well work by simply replacing self.headers by self.dataframe.columns in the generate_preview_info_from_rows_results() function. We would have to check if it does not break something else (experiment_run) but it would be more logical to have all input columns in the report.

Reply:
I think it might be better instead of replacing them too add the columns that are not part of the header.